plugins/registry.ts

```diff
@@ -342,6 +344,29 @@ export default class PluginRegistry {
         return id;
     });
 
+    // Register a component to render a custom file attachments in posts.
+    // Accepts the following:
+    // - match - A function that receives fileInfo and returns a
+    //   boolean indicating if the plugin is able to process it.
+    // - component - The component that renders the attachment preview
+    // Returns a unique identifier.
+    registerFileAttachmentPluginComponent = reArg(['component', 'match'], ({match, component}) => {
+        const id = generateId();
+
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_COMPONENT,
+            name: 'FileAttachmentPluginComponent',
+            data: {
+                id,
+                pluginId: this.id,
+                component: fileAttachmentConnector(component as ComponentType<FileAttachmentProps>),
```

Comment:
Why exactly is this necessary, versus just using `component` here? Not sure what's going on here